src/agentconfig.rs

```diff
@@ -310,6 +416,18 @@ impl Configurable for AgentConfig {
             None
         };
 
+        let mut final_port = if let Ok(server_port) =
+            AgentConfig::get_exactly_one_string(&parsed_values, &AgentConfig::SERVER_PORT)
+        {
+            u16::from_str(&server_port)
+                .unwrap_or_else(|_| panic!("Error parsing webserver port from [{}], aborting."))
+        } else {
+            // This should not be necessary, as the default should be provided by clap,
```

Comment:
Same as above. If it shouldn't happen let's panic?

Reply:
I've refactored the code and moved this bit out into a fn.